presto-main/src/main/java/com/facebook/presto/util/ImmutableCollectors.java

```diff
@@ -74,4 +74,16 @@ private ImmutableCollectors() {}
                 },
                 ImmutableMap.Builder::build);
     }
+
+    public static <K, V> Collector<K, ?, ImmutableMap<K, V>> toImmutableMapReversed(Function<K, V> keyMapper)
```

Comment:
The name of this method is weird and confusing. Also, I don't think you need the method at all (see my comment above on how to structure the code in the LogicalPlanner)